### DISCUSSION TITLE
Why change from static properties to named exports for the new data-fetching methods?

### DISCUSSION THREAD

**Discussion by rafaelalmeidatk** [vercel/next.js#12184] [Help] [answered]
I am not criticizing the change, just curious about the reason since `getInitialProps` is a static property in the page component and now we have named exports for the new methods. Is it for code elimination (libs utilized only in server-side methods, for example), easier to type, or perhaps a more semantic usage of modules? (I am not sure this last thing I said makes sense lol)

**Comment by timneutkens** [comment, marked as answer]
All of the above 😄

- It's impossible to do the code elimination without having it as an explicit export as there are many ways the property could be assigned.
- With class components the static method made sense semantically but since hooks were introduced everyone is more likely to write function components where the assignment was slightly strange
- It's easier to type and extract the type of `getStaticProps`/`getServerSideProps` as they're not mixing with the React types

**Reply by rafaelalmeidatk** [reply]
Thanks for the answer, Tim!
